### ISSUE TITLE
Sidekiq isolator doesn't work if "sidekiq/testing" is required only AFTER "isolator"

### ISSUE THREAD

**Issue by timdiggins** [palkan/isolator#52]
## Is your feature request related to a problem? Please describe.

In a project I'm migrating from DelayedJob under ActiveRecord to raw Sidekiq (ie not through ActiveJob). I've discovered that I've missed a few areas where I forgot to consider done after_commit. 

I installed this gem hoping it would pick up on them, It took me a while to realize that although it has a Sidekiq module, it only supports the calls for ActiveJob api (`deliver_later`). 

It also (questionably!) seems to support  a call that is I think DelayedJob specific (e.g. xxx.delay https://github.com/palkan/isolator/blob/master/spec/isolator/adapters/background_jobs/sidekiq_spec.rb#L24)

## Describe the solution you'd like

I would like the gem to raise an error with `SidekiqJobClass.perform_async` when in a transaction

## Additional context

While doing this it might be good to modernize terminology in the rspec test - `Worker` -> `Job` (https://github.com/mperham/sidekiq/discussions/4971)

**Comment by timdiggins** [comment]
Oops I've just looked at the sidekiq spec in more details (starting on a PR) and see that it does actually support sidekiq native (it's just that the descriptions don't correspond with the code).

So maybe this is a bug report, but I haven't yet worked out why it's not raising an error in my project. I'll see if I can work out a minimal test case

**Comment by timdiggins** [comment]
I'm closing this as I got the wrong end of the stick - will create a new issue
